scripts/keypoint_regression.py

```diff
@@ -14,81 +17,132 @@
 
 
 @dataclass
 class AppSettings(Serializable):
     model: KeypointNetwork2D.Settings = KeypointNetwork2D.Settings()
-    data: ColoredCubeDataset.Settings = ColoredCubeDataset.Settings()
+
+    # Select dataset among different options, by name.
+    dataset: DatasetOptions = DatasetOptions.OBJECTRON
+
+    cube: ColoredCubeDataset.Settings = ColoredCubeDataset.Settings()
+    objectron: Objectron.Settings = Objectron.Settings()
+    sample_objectron: SampleObjectron.Settings = SampleObjectron.Settings()
+
     # NOTE(ycho): root run path is set to tmp dir y default.
     path: RunPath.Settings = RunPath.Settings(root='/tmp/ai604-kpt')
     train: Trainer.Settings = Trainer.Settings()
+    batch_size: int = 8
     device: str = ''
 
 
-def load_data(opts: AppSettings):
+def load_data(opts: AppSettings, device: th.device):
     """ Fetch pair of (train,test) loaders for MNIST data """
-    # FIXME(ycho): Incorrect data loading scheme, fix.
-    # Configure loaders.
-    transform = transforms.Compose([
-        transforms.ToTensor(),
-        transforms.Normalize((0.1307,), (0.3081,))
+
+    # get_dataset(test:bool)
+
+    # TODO(ycho): Consider scripted compositions?
+    transform = Compose([
+        DenseMapsMobilePose(DenseMapsMobilePose.Settings(), device),
+        Normalize(Normalize.Settings())
     ])
-    train_dataset = datasets.MNIST(opts.data_dir, train=True, download=True,
-                                   transform=transform)
-    test_dataset = datasets.MNIST(opts.data_dir, train=False, download=False,
-                                  transform=transform)
+
+    # TODO(ycho): Prefer unified interface across dataset instances.
+    if opts.dataset == DatasetOptions.CUBE:
+        data_opts = opts.cube
+        train_dataset = ColoredCubeDataset(
+            data_opts, device, transform)
+        test_dataset = ColoredCubeDataset(
+            data_opts, device, transform)
+    elif opts.dataset == DatasetOptions.OBJECTRON:
+        data_opts = opts.objectron
+
+        data_opts = replace(data_opts, train=True)
+        train_dataset = Objectron(data_opts, transform)
+
+        data_opts = replace(data_opts, train=False)
+        test_dataset = Objectron(data_opts, transform)
+    elif opts.dataset == DatasetOptions.SAMPLE_OBJECTRON:
+        # TODO(ycho): Maybe not the most elegant idea.
+        # The code looks complex due to the nested replacement.
+        objectron_opts = replace(
+            opts.sample_objectron.objectron, train=True)
+        data_opts = replace(
+            opts.sample_objectron, objectron=objectron_opts)
+        train_dataset = SampleObjectron(data_opts, transform)
+
+        objectron_opts = replace(
+            opts.sample_objectron.objectron, train=False)
+        data_opts = replace(
+            opts.sample_objectron, objectron=objectron_opts)
+        test_dataset = SampleObjectron(data_opts, transform)
+    else:
+        raise ValueError(F'Invalid dataset choice : {opts.dataset}')
```

Comment:
I'm not familiar with dataclass yet, so could you please explain what happens with `replace()`?

Reply:
If there's a setting that's like `s={"a":1, "b":2}`, then `replace(s, a=3)` would make the resulting output `s={"a":3, "b":2}`. `replace` is required since generally we consider the `dataclass` instance itself to be immutable. This piece of code *is* quite confusing so I'll leave comments for this.

Reply:
Oh, I got it, thank you.